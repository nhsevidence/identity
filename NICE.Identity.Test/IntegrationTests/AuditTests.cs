using NICE.Identity.Models;
using NICE.Identity.Test.Infrastructure;
using System;
using System.Globalization;
using System.Threading.Tasks;
using Xunit;

namespace NICE.Identity.Test.IntegrationTests
{
	public class AuditTests : TestBase
	{
		[Fact]
		public async Task GetAudit()
		{
			//Arrange
			var context = GetContext();
			context.Audit.Add(new Audit("User", "FirstName", "Peter", "Phil", DateTime.ParseExact("03-Jan-2019 10:25", "dd-MMM-yyyy hh:mm", DateTimeFormatInfo.InvariantInfo), Guid.Empty));
			context.SaveChanges();
			var client = GetClient(context);

			//Act
			var response = await client.GetAsync("/admin/api/Audit");
			response.EnsureSuccessStatusCode();
			var responseString = await response.Content.ReadAsStringAsync();

			//Assert
			responseString.ShouldMatchApproved(new Func<string, string>[]{ Scrubbers.ScrubAuditId });
		}
	}
}
