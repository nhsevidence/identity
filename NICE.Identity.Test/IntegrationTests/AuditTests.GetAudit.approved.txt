[
    {
        "auditId":"scrubbed by ScrubAuditId",
        "tableName":"User",
        "fieldName":"FirstName",
        "oldValue":"Peter",
        "newValue":"Phil",
        "date":"2019-01-03T10:25:00",
        "userId":"00000000-0000-0000-0000-000000000000"
    }
]